platform/core/src/services/ViewportGridService/ViewportGridService.ts

```diff
@@ -105,8 +105,8 @@ class ViewportGridService extends PubSubService {
     this.setDisplaySetsForViewports([props]);
   }
 
-  public setDisplaySetsForViewports(props) {
-    this.serviceImplementation._setDisplaySetsForViewports(props);
+  public async setDisplaySetsForViewports(props) {
+    await this.serviceImplementation._setDisplaySetsForViewports(props);
     const state = this.getState();
     const viewports = [];
 
```

Comment:
I didn't expect this async method here, why?

Reply:
That is a `dispatch` method from `ViewportGridProvider` that runs asynchronously and without `await` this method was returning the old state (w/ old `displaySetInstanceUID`) instead of the updated one.